pkg/scalers/kafka_scaler.go

```diff
@@ -430,17 +436,31 @@ func (s *kafkaScaler) getTopicOffsets(partitions []int32) (map[int32]int64, erro
 		request.AddBlock(s.metadata.topic, partitionID, sarama.OffsetNewest, 1)
 	}
 
-	offsets := make(map[int32]int64)
-
 	// Step 2: send requests, one per broker, and collect offsets
+	resultCh := make(chan brokerOffsetResult, len(requests))
+	var wg sync.WaitGroup
+	wg.Add(len(requests))
 	for broker, request := range requests {
-		response, err := broker.GetAvailableOffsets(request)
+		go func(brCopy *sarama.Broker, reqCopy *sarama.OffsetRequest) {
+			response, err := brCopy.GetAvailableOffsets(reqCopy)
+			resultCh <- brokerOffsetResult{response, err}
+			wg.Done()
+		}(broker, request)
```

Comment:
I'm not a super expert in golang (neither in golang concurrency) but I think that in this case we should use `defer` to ensure the call.
```suggestion
		go func(brCopy *sarama.Broker, reqCopy *sarama.OffsetRequest) {
			defer wg.Done()
			response, err := brCopy.GetAvailableOffsets(reqCopy)
			resultCh <- brokerOffsetResult{response, err}
		}(broker, request)
```
WDYT?

Reply:
`defer` will ensure the statement will be executed whenever the func returns;
in this case, because there is no other return condition, so putting done in defer or not would behave the same.

from readability point of view having defer could be a plus though.

Reply:
Agreed, defer would help with readability. Committed your suggestion 👍